lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2384,12 +2404,14 @@ NamingPatternRequest::evaluate(Evaluator &evaluator, VarDecl *VD) const {
   }
 
   if (!namingPattern) {
-    // Try type checking parent conditional statement.
+    // Try type checking parent control statement.
     if (auto parentStmt = VD->getParentPatternStmt()) {
-      if (auto LCS = dyn_cast<LabeledConditionalStmt>(parentStmt)) {
-        TypeChecker::typeCheckConditionForStatement(LCS, VD->getDeclContext());
```

Comment:
Now `typeCheckConditionForStatement()` can be moved from `TypeCheckConstraints.cpp` to `TypeCheckStmt.cpp`, and be internalized in the file. I will do it in a follow up PR.